### DISCUSSION TITLE
Caption under Image

### DISCUSSION THREAD

**Discussion by bryanxhuang** [markdoc/markdoc#175] [Q&A] [answered]
I've been trying to make a node for the image markdown component to take in a caption. I am wondering if there's a better way to implement this or if captions could be included in the new roadmap features for Image.

Markdown code looks something like this:
```
![this is a flamingo](https://images.unsplash.com/photo-1490750967868-88aa4486c946?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1200&q=80 "caption of the picture")
```

My component looks something like this. I originally used `<figcaption>` for the html element, but since the image markdown was nested in a `<p>` tag and giving bad errors, I just hacked something together with the `<a>` tag. 

```
export function Galla({ src, title }) {
    return (
        <>
            <img src={src} layout='fill'>
            </img>
            <a>{title}</a>
        </>
    );
}
```

It works, but feel like there's gotta be a better way to do this than an `<a>` tag.

 I'm very new to markdoc and just a curious student. Apologize if there's an obvious solution to this or this is tagged wrong.

**Comment by mfix-stripe** [comment, marked as answer]
Hey @bryanxhuang, we have a very similar use case at Stripe, but we use an image _tag_ to handle it, like this:
```md
{% image src="/image" alt="" %}
Image caption goes here
{% /image %}
```

That should work for you!

**Reply by bryanxhuang** [reply]
Thanks, just implemented and works great. Appreciate all the great work here :)

**Reply by mfix-stripe** [reply]
Thanks for saying so @bryanxhuang 🙏
